examples/complete-hosted-zone-in-diff-aws/main.tf

```diff
@@ -0,0 +1,64 @@
+provider "aws" {
+  region = "us-east-1"
+}
+
+# this is where hosted zone present/need to be created
+provider "aws" {
+  region = "us-east-1"
+  alias  = "production"
+}
+
+
+locals {
+  # Use existing (via data source) or create new zone (will fail validation, if zone is not reachable)
+  use_existing_route53_zone = true
+
+  domain = "terraform-aws-modules.modules.tf"
+
+  # Removing trailing dot from domain - just to be sure :)
+  domain_name = trimsuffix(local.domain, ".")
+}
+
+data "aws_route53_zone" "this" {
+  provider = aws.production
+
+  count = local.use_existing_route53_zone ? 1 : 0
+
+  name         = local.domain_name
+  private_zone = false
+}
+
+resource "aws_route53_zone" "this" {
+  provider = aws.production
+
+  count = ! local.use_existing_route53_zone ? 1 : 0
+  name  = local.domain_name
+}
+
+module "acm" {
+  source = "../../"
+
+  domain_name = local.domain_name
+  zone_id     = coalescelist(data.aws_route53_zone.this.*.zone_id, aws_route53_zone.this.*.zone_id)[0]
+
+  subject_alternative_names = [
```

Comment:
I suggest changing the `subject_alternative_names` to:

```
  subject_alternative_names = [
    "dev.${local.domain_name}",
    "*.dev.${local.domain_name}",
  ]
```

to remain consistent with your example use case of a prod account holding the Route 53 zone and a dev account creating a cert for some domains within the zone.